src/TinyContainerBuilder.php

```diff
@@ -167,4 +167,43 @@ private function isAbsolute(string $path)
     {
         return (new Filesystem())->isAbsolutePath($path);
     }
+
+    public function excludeSourcePath(string $excludePath): ContainerBuilderInterface
+    {
+        if (!$this->isAbsolute($excludePath)) {
+            if (!isset($this->rootPath)) {
+                throw new \LogicException(
+                    \sprintf('When relative path is provided root should be set first. Privided: %s', $excludePath)
+                );
+            }
+            $excludePath = rtrim($this->rootPath, '/') . '/' . $excludePath;
+        }
+        if (!\file_exists($excludePath)) {
+            throw new \RuntimeException(\sprintf('Provided exclude path is not a valid pathname: %s', $excludePath));
+        }
+
+        // The extra slash at the end prevents exclusion of sibling paths starting with exclude path name
+        // For example /usr/bin/php shouldn't exclude /usr/bin/php7.4
+        $excludePath = $this->removeRelativePathParts($excludePath) . '/';
+        $this->paths = array_filter($this->paths, function (string $path) use ($excludePath) {
+            return 0 !== stripos($this->removeRelativePathParts($path) . '/', $excludePath);
```

Comment:
it's probably a little more efficient to do this to each added path as it gets added (in add path) instead of every time we exclude a path. but in most of our applications, that's probably a micro-optimization.

Reply:
I am enforcing it now. I resolve relative path parts when making a path absolute.